AssetMappr/application/submit_new_asset_cb.py

```diff
@@ -0,0 +1,86 @@
+"""
+File: submit_new_asset_cb.py
+Author: Mihir Bhaskar
+
+Desc: This file creates the callbacks which interact with the submit_new_asset popup
+      
+      This is linked to the submit_new_asset.py layout file, as well as the submit_new_asset_db
+      file in the Database folder, which writes the new user-entered asset info to the database
+
+Input: 
+    app: an initialized dash app
+    
+Output: 
+    Callbacks relating to the submit-new-asset feature
+     
+"""
+import dash
+from dash.dependencies import Input, Output, State
+from dash import dash_table
+from dash import dcc
+from dash import html
+import pandas as pd
+import dash_leaflet as dl
+
+from flask import request
+
+from AssetMappr.database.submit_new_asset_db import submit_new_asset_db
+
+
+def submit_new_asset_cb(app):
+    
+    # Callback to interact with the open and close buttons of the modal
+    @app.callback(
+        Output('submit-asset-modal', 'is_open'),
+        [Input('open-asset-submit', 'n_clicks'), Input('close-asset-submit', 'n_clicks')],
+        [State('submit-asset-modal', 'is_open')],
+        )
+    def toggle_modal(n1, n2, is_open):
+        if n1 or n2:
+            return not is_open
+        return is_open
+    
+    # Callback to render the Leaflet map on which users will pin the location of the asset
+    @app.callback(
+        Output('submit-asset-map', 'children'),
+        Input('submit-asset-modal', 'is_open')
+        )
+    def render_map_on_show(is_open):
+        if is_open:
+            return dl.Map([dl.TileLayer(), dl.LayerGroup(id='layer')],
+                      id='submit-asset-map', 
+                      zoom=14, center=(39.8993885, -79.7249338),
```

Comment:
Sure you are aware of this, but flagging that eventually this centering should be based on some sort of user input (select which community you are in), or automatic location tracking. Might suggest the former.

Reply:
Adding a TODO comment to this line of code to change it for later. The idea is that users will first enter which community they are in, then the rest of the app will load (including the 'center' coordinates that we decided to store in the communities_master table, a filtered version of the assets table for just that community, etc.)